src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -280,6 +281,7 @@ private int[] findTopApproximatePostings(float[] queryVector, IntArrayList segme
         // top K ascending
         for (int i = end; i >= 0; i--)
             postings[end - i] = pairs.get(i).node;
+        Arrays.sort(postings);
```

Comment:
comment pls

Reply:
i'm actually surprised anything works w/o this, is something else sorting higher up the stack?

Reply:
There is no sorting after this point. I think the only failure mode is for shadowed keys because those only get deduped in the `RangeUnionIterator` when this collection is sorted. Also, we have this code that attempts to dedupe before reading from storage, and that wouldn't work if we return out of order:

https://github.com/datastax/cassandra/blob/ca130910e2372faad6815ff5e9dd873f9cbf3c78/src/java/org/apache/cassandra/index/sai/plan/StorageAttachedIndexSearcher.java#L500-L502

However, sorting mostly appears to be helpful for limiting duplicate reads, not for functional correctness. Even the shadowed key problem is somewhat arbitrary--we don't have to assert that we don't get dupes, though it does seem good for ensuring good performance.